interfaces/cython/cantera/base.pyx

```diff
@@ -232,6 +232,21 @@ cdef class _SolutionBase:
         def __get__(self):
             return anymap_to_dict(self.base.parameters(True))
 
+    def update_user_data(self, data):
+        """
+        Add the contents of the provided `dict` as additional fields when generating
+        YAML phase definition files with `write_yaml` or in the data returned by
+        `input_data`. Existing keys with matching names are overwritten.
```

Comment:
The docstring for `AnyMap::update` says that

> If keys in `other` also exist in this AnyMap, the `keepExisting` option determines which item is used.

By my reading, setting `keepExisting` as `True` means that the existing keys are kept and duplicate keys are _not_ overwritten. Can you clarify?

Reply:
Good catch -- I had this backwards. The tests have been updated to verify this as well.